core/src/main/java/org/elasticsearch/index/query/QueryParsingException.java

```diff
@@ -44,7 +44,8 @@ public QueryParsingException(QueryParseContext parseContext, String msg, Object.
     }
 
     public QueryParsingException(QueryParseContext parseContext, String msg, Throwable cause, Object... args) {
-        super(parseContext.index(), msg, cause, args);
+        super(msg, cause, args);
```

Comment:
we use this super(...); addIndex(); pattern often. I wonder if it makes sense to add an ElasticsearchException ctor which takes an index as an argument and does this for you. Same goes for shardIds..